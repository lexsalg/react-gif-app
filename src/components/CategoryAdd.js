import { useState } from "react";

import PropTypes from 'prop-types';

export const CategoryAdd = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            // setCategories([...categories, 'nuevo'])
            setCategories(state => [inputValue, ...state])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>

            <h1>{inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}


CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}