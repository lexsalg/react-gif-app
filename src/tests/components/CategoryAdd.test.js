import { shallow } from 'enzyme';

import { CategoryAdd } from '../../components/CategoryAdd'

describe('Pruebas en <CategoryAdd />', () => {

    // const setCategories = () => { };
    const setCategories = jest.fn();

    let wrapper = shallow(<CategoryAdd setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
    })


    test('should be displayed correctly', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('should changed text box', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);

    })


    test('No debe de postear la informaciÃ³n con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } })

        expect(setCategories).not.toHaveBeenCalled();

    })


    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Hola mundo';

        const preventDefault = () => { };

        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');

    })



})
