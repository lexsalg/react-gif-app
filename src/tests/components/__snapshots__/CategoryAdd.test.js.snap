// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CategoryAdd /> should be displayed correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <h1 />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
